### PR TITLE
Switch to setuptools

### PR DESCRIPTION
**Pull request by opadron**
Building wheels should now work!

Also refactors some of the test code.  Subproject tests (defined in their `CMakeLists.txt` files) are currently commented out.  They will later be replaced with nose-friendly versions.

### CODE REVIEW

**Comment by opadron** [comment]
This should fix #85

**Comment by thewtex** [comment]
woot!  LGTM, but CI is not yet passing

**Comment by opadron** [comment]
@thewtex could you take another pass at this?  I've added some more changes to get CI working.

File: skbuild/cmaker.py [hunk 1]
```diff
@@ -103,24 +102,45 @@ def configure(self, clargs=(), generator_id=None):
...
-               '-DPYTHON_INCLUDE_DIR:PATH=' + python_include_dir,
-               '-DPYTHON_LIBRARY:FILEPATH=' + python_library,
-               '-DSKBUILD:BOOL=TRUE',
-               "-DCMAKE_MODULE_PATH:PATH={}".format(
-                   os.path.dirname(__file__) + '/resources/cmake')
-               ]
+        cwd = os.getcwd()
+        cmd = ['cmake', cwd, '-G', generator_id]
+        cmd.extend(
+            "=".join((cmake_variable, value)) for cmake_variable, value in (
```
Comment (jcfr) [inline skbuild/cmaker.py]: Could we simply extend the list, I don't think using the join and for loop give us much here ?

File: skbuild/cmaker.py [hunk 2]
```diff
@@ -151,27 +171,61 @@ def get_python_version():
...
-                os.path.join(sysconfig.get_python_inc(),
-                             ".".join(map(str, sys.version_info[:2]))),
-                sysconfig.get_python_inc()
-                ]
+        if not found_python_h:
+            candidate_prefixes = []
+
+            # NOTE(opadron): these possible prefixes must be guarded against
+            # AttributeErrors and KeyErrors because they each can throw on
+            # different platforms or even different builds on the same platform.
```
Comment (jcfr) [inline skbuild/cmaker.py]: Since the documentation indicate `None` is returned, could you revert these changes ?

* https://docs.python.org/2/library/sysconfig.html#sysconfig.get_config_var
* https://docs.python.org/3/library/sysconfig.html#sysconfig.get_config_var

On Linux, no error:
```
$ python -c "import sysconfig; print(sysconfig.get_config_var('PYTHONPATH'))"
:plat-x86_64-linux-gnu:lib-tk:lib-old

$ python -c "import sysconfig; print(sysconfig.get_config_var('Invalid'))"
None
```

On MacOSx, no error:
```
$ python -c "import sysconfig; print(sysconfig.get_config_var('Invalid'))"
None
$ python -c "import sysconfig; print(sysconfig.get_config_var('PYTHONPATH'))"
:plat-darwin:plat-mac:plat-mac/lib-scriptpackages:../../Extras/lib/python:lib-tk:lib-old
```

On windows, I expect the same.

File: skbuild/setuptools_wrap.py [hunk 3]
```diff
@@ -59,19 +63,21 @@ def parse_args():
                              concatenate_value=True)
     dutils, cmake = move_arg('-G', dutils, cmake)
     dutils, make = move_arg('-j', dutils, make)
-    op = os.path
-
-    def absappend(x):
-        return op.join(op.dirname(op.abspath(sys.argv[0])), x)
+    absappend = (
+        lambda x: os.path.join(
```
Comment (jcfr) [inline skbuild/setuptools_wrap.py]: Do not use lambda here. See https://github.com/scikit-build/scikit-build/commit/1cd88e7b24cfe03fb06472de8a6619cc38f96439

**Comment by jcfr** [comment]
xref #63 (Refactor testing of scikit-build sample projects)

File: skbuild/cmaker.py (hunk 1 above)
Comment (opadron) [inline skbuild/cmaker.py]: Different strokes, I guess.  I had hoped this would make the arguments that we're passing easier to read through.

File: skbuild/cmaker.py (hunk 2 above)
Comment (opadron) [inline skbuild/cmaker.py]: You're right, that particular piece does not need to be guarded.

File: skbuild/setuptools_wrap.py (hunk 3 above)
Comment (opadron) [inline skbuild/setuptools_wrap.py]: Any particular reason?

Comment (blowekamp) [inline skbuild/setuptools_wrap.py]: JC provided a link in the patch linked:
http://stackoverflow.com/questions/25010167/e731-do-not-assign-a-lambda-expression-use-a-def

PEP-8

Comment (opadron) [inline skbuild/setuptools_wrap.py]: @blowekamp I see, now.  I didn't notice that link.  Makes sense to me.

**Comment by opadron** [comment]
@scikit-build/developers PTAL

**Comment by thewtex** [comment]
:+1:

**Comment by opadron** [comment]
Wheel builders, rejoice! 🎉

**Comment by jcfr** [comment]
It turns out that this topic broke the submission of `coverage.xml`, it is now generated in this folder:

```
./tests/samples/fail-outside-project-root/coverage.xml
```

instead of 

```
./tests/coverage.xml
```

**Comment by jcfr** [comment]
It turns out that the test `outside_project_root` is causing the `coverage.xml` to be generated in "wrong" location
